### PR TITLE
[HOOTL] Verify url secret

### PR DESCRIPTION
**Pull request by pmilliotte**
## Description

Adding doc-ack label as we will add documentation once opening endpoint to public

After having added a "urlSecret" property to webhook source, this PR displays it in the webhook url, and verify for incoming webhooks that the secret in as a query param matches the webhook source's one.

Screenshot 2025-10-06 at 11 18 09

Response when secret does not match :
```
{
    "error": {
        "type": "not_authenticated",
        "message": "Invalid webhook URL secret."
    }
}
```

Response when it matches :
```
{
    "success": true
}
```

## Tests

Added a test with unmatching url secret

## Risk

low

## Deploy Plan

- Block front deploy ❌
- merge PR
- run backfill script in us + eu
- run migration in us + eu
- deploy to front
- Unblock front deploy ✅

### CODE REVIEW

File: front/components/triggers/WebhookSourceDetailsInfo.tsx [hunk 1]
```diff
@@ -68,8 +68,12 @@ export function WebhookSourceDetailsInfo({
 
   const webhookUrl = useMemo(() => {
     const { url } = config.getDustAPIConfig();
-    return `${url}/api/v1/w/${owner.sId}/triggers/hooks/${webhookSourceView.webhookSource.sId}`;
-  }, [owner.sId, webhookSourceView.webhookSource.sId]);
+    return `${url}/api/v1/w/${owner.sId}/triggers/hooks/${webhookSourceView.webhookSource.sId}?secret=${webhookSourceView.webhookSource.urlSecret}`;
```
[the remark this answers is not in the document]
Comment (adrsimon) [reply front/components/triggers/WebhookSourceDetailsInfo.tsx]: we should maybe not use `?secret=$secret` but just `/$secret` ?
hard code in the url and not in a param

File: front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/index.test.ts [hunk 2]
```diff
@@ -60,6 +60,7 @@ describe("POST /api/v1/w/[wId]/triggers/hooks/[webhookSourceId]", () => {
     req.query = {
       wId: workspace.sId,
       webhookSourceId: webhookSource.sId(),
+      secret: webhookSource.urlSecret ?? undefined,
```
Comment (adrsimon) [inline front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/index.test.ts]: urlSecret is nullable ?

File: front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/index.ts [hunk 3]
```diff
@@ -143,6 +143,20 @@ async function handler(
     });
   }
 
+  // Validate webhook url secret
+  if (
+    typeof query.secret !== "string" ||
+    query.secret !== webhookSource.urlSecret
```
Comment (adrsimon) [inline front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/index.ts]: we store the hardcoded version right ? should we store a hash of the secret and check against the hash ? 

prevents us from any db leak / extraction

File: front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/[webhookSourceUrlSecret]/index.test.ts [hunk 4]
```diff
@@ -247,4 +250,126 @@ describe("POST /api/v1/w/[wId]/triggers/hooks/[webhookSourceId]", () => {
     expect(data.error.type).toBe("not_authenticated");
     expect(data.error.message).toBe("Invalid webhook signature.");
   });
+
+  it("returns 401 when webhook URL secret is invalid", async () => {
```
Comment (fabiencelier) [inline front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/[webhookSourceUrlSecret]/index.test.ts:257]: I'm not sure it is possible to create a webhook with empty secret, but if it is, can we also do a test with empty secret in DB + empty secret in query = 200 ?

File: front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/index.ts (hunk 3 above)
Comment (fabiencelier) [reply front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/index.ts]: We still need to display it back to the user for 10 minutes, so the implem might be a bit harder (we need to store the actual secret somewhere then revoke it)
But I agree storing the real secret forever is not ideal in term of security

File: front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/index.test.ts (hunk 2 above)
Comment (pmilliotte) [reply front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/index.test.ts]: @adrsimon 

**About query param**: I used it to keep using the same endpoint. It is not a big refactor though, so taking the comment into account ✅

**About nullable urlSecret**: I made it nullable for now to avoid a backfill script in prod. I have the feeling we are not 100% confident about the current implem, so what about leaving it nullable for now and backfilling when opening to public ? 🤔

File: front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/index.ts (hunk 3 above)
Comment (pmilliotte) [reply front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/index.ts]: @adrsimon @fabiencelier I don't see the point since we generate this secet. So saving a hash or a generated secret won't differ, because in any cases we display it (the hash or the generated secret) to the admin in the url...

File: front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/[webhookSourceUrlSecret]/index.test.ts (hunk 4 above)
Comment (pmilliotte) [reply front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/[webhookSourceUrlSecret]/index.test.ts:257]: @fabiencelier type of urlSecret is nullable to avoid running backfill script in prod for now. In practice it is never null. I have added a test when pathParam is null

Curious of what you think about waiting a bit before running a "non nullable" migration + backfill

Comment (fabiencelier) [reply front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/[webhookSourceUrlSecret]/index.test.ts:257]: Ok, it makes sense :)

I don't know if it is used by anybody as the moment, if it 's just us I think we can do the migration script soon
But I don't know what a good default secret will be :/

File: front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/index.test.ts (hunk 2 above)
Comment (pmilliotte) [reply front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/index.test.ts]: Seen IRL doing the backfill + migration now

File: front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/[webhookSourceUrlSecret]/index.test.ts (hunk 4 above)
Comment (pmilliotte) [reply front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/[webhookSourceUrlSecret]/index.test.ts:257]: @fabiencelier as seen IRL with adrien, I do the migration now and generate a secret for existing webhooks

**Review by adrsimon — APPROVED** [review]

File: front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/index.ts (hunk 3 above)
Comment (fabiencelier) [reply front/pages/api/v1/w/[wId]/triggers/hooks/[webhookSourceId]/index.ts]: What about we store the secret and the hash in diffrent columns.
We store the hash permanently and use it for validating the input secret
We only store and display the secret for 10 minutes, then we remove it / hide all chars but the last 4 

This depends on whether we want to display the URL with secret forever or just for 10 minutes
I'm totally ok to delay this discussion to later and merge the PR as is
